### PR TITLE
bugfix trace ut & data race

### PR DESCRIPTION
**Pull request by xzxiong**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3706

## What this PR does / why we need it:
- fix failed ut with timer in `batch_processor_test.go`, which have a little random problem
- fix data race in `batch_processor.go`, which is not found in previous CI.

### CODE REVIEW

File: pkg/util/export/batch_processor.go [hunk 1]
```diff
@@ -90,32 +90,28 @@ func (b *bufferHolder) Add(item batchpipe.HasName) {
...
 }
 
-// Stop set bufferHolder readonly, and hold Add request
-func (b *bufferHolder) Stop() bool {
+// StopAndGetBatch set bufferHolder readonly, which can hold Add request,
+// and gen batch request content
+func (b *bufferHolder) StopAndGetBatch(buf *bytes.Buffer) bool {
 	b.mux.RLock()
 	defer b.mux.RUnlock()
 	if !atomic.CompareAndSwapUint32(&b.readonly, READWRITE, READONLY) {
```
Comment (fengttt) [inline pkg/util/export/batch_processor.go:98]: So b.mux will protect the data structure?    In that case do we really need atomic?

Comment (xzxiong) [reply pkg/util/export/batch_processor.go:98]: seems a little bit redundant, i will reconsider it.

Comment (xzxiong) [reply pkg/util/export/batch_processor.go]: fixed, just use `RWMutex` as protection.

**Review by fengttt — APPROVED** [review]

**Comment by xzxiong** [comment]
@mergify refresh

**Pull request merged**